std-bits/table/src/main/java/org/enso/table/parsing/DateTimeParser.java

```diff
@@ -17,11 +17,13 @@ public DateTimeParser(String[] formats, Locale locale) {
 
   private static ZonedDateTime parse(String text, DateTimeFormatter formatter)
       throws DateTimeParseException {
-    try {
-      return ZonedDateTime.parse(text, formatter);
-    } catch (DateTimeParseException ignored) {
-      return LocalDateTime.parse(text, formatter).atZone(ZoneId.systemDefault());
+    var datetime = formatter.parseBest(text, ZonedDateTime::from, LocalDateTime::from);
+    if (datetime instanceof ZonedDateTime zdt) {
+      return zdt;
+    } else if (datetime instanceof LocalDateTime ldt) {
+      return ldt.atZone(ZoneId.systemDefault());
     }
+    throw new DateTimeParseException("Invalid date time", text, 0);
```

Comment:
In hindsight, I don't know how I thought that throwing exceptions for every parsed date is good design... Well possibly I just didn't think of any other possible solution, but wow this would have never been efficient - glad that you figured out a solution without exceptions!